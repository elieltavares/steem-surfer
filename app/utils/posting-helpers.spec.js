/* eslint-disable */
import {
  createPermlink,
  makeOptions,
  extractMetadata,
  makeJsonMetadata,
  createReplyPermlink,
  makeJsonMetadataReply,
  createPatch
} from './posting-helpers';
import { Select } from 'antd/lib/select';
import React from 'react';

describe('createPermlink', () => {
  it('(1) ', () => {
    const input = 'lorem ipsum dolor sit amet';
    expect(createPermlink(input)).toMatchSnapshot();
  });
});

describe('createPermlink random', () => {
  jest.spyOn(Math, 'random').mockImplementation(() => {
    return 1.95136022969379;
  });

  it('(1) ', () => {
    const input = 'lorem ipsum dolor sit amet';
    expect(createPermlink(input, true)).toMatchSnapshot();
  });
});

describe('createPermlink non-latin chars', () => {
  jest.spyOn(Math, 'random').mockImplementation(() => {
    return 1.95136022969379;
  });

  it('(1) ', () => {
    const input = 'ปลาตัวใหญ่สีเหลืองทอง';
    expect(createPermlink(input)).toMatchSnapshot();
  });
});

describe('makeOptions', () => {
  it('(1) Default 50% / 50%', () => {
    expect(
      makeOptions('talhasch', 'lorem-ipsum-1', 'default')
    ).toMatchSnapshot();
  });

  it('(2) Power Up 100%', () => {
    expect(makeOptions('talhasch', 'lorem-ipsum-1', 'sp')).toMatchSnapshot();
  });

  it('(3) Decline Payout', () => {
    expect(makeOptions('talhasch', 'lorem-ipsum-1', 'dp')).toMatchSnapshot();
  });
});

describe('extractMetadata', () => {
  it('(1) ', () => {
    const input = '<img src="http://www.xx.com/a.png"> @lorem @ipsum';
    expect(extractMetadata(input)).toMatchSnapshot();
  });

  it('(2) ', () => {
    const input =
      '@lorem <img src="http://www.xx.com/a.png"> ![h74zrad2fh.jpg](https://img.esteem.ws/h74zrad2fh.jpg) http://www.google.com/foo/bar  @ipsum';
    expect(extractMetadata(input)).toMatchSnapshot();
  });
});

describe('makeJsonMetadata', () => {
  it('(1) ', () => {
    const meta = {
      image: [
        'http://www.xx.com/a.png',
        'https://img.esteem.ws/h74zrad2fh.jpg'
      ],
      links: ['http://www.google.com/foo/bar'],
      users: ['lorem', 'ipsum']
    };
    const tags = ['esteem', 'art'];

    expect(makeJsonMetadata(meta, tags, '2.0.0')).toMatchSnapshot();
  });
});

describe('makeJsonMetadataReply', () => {
  it('(1)', () => {
    expect(makeJsonMetadataReply(['foo', 'bar'])).toMatchSnapshot();
  });
});

describe('createReplyPermlink', () => {
  jest.spyOn(Date, 'now').mockImplementation(() => {
    return new Date('2018-09-21T12:00:50.000Z');
  });

  it('(1) ', () => {
    expect(createReplyPermlink('good-karma')).toMatchSnapshot();
  });
});

describe('createPatch', () => {
  it('(1) ', () => {
    expect(
      createPatch('lorem ipsum dlor sit amet', 'lorem ipsum dolor sit amet')
    ).toMatchSnapshot();
  });
});
