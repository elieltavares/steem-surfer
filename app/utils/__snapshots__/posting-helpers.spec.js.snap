// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createPatch (1)  1`] = `
"@@ -6,16 +6,17 @@
  ipsum d
+o
 lor sit 
"
`;

exports[`createPermlink (1)  1`] = `"lorem-ipsum-dolor-sit-amet"`;

exports[`createPermlink non-latin chars (1)  1`] = `"f38c58113ff7e"`;

exports[`createPermlink random (1)  1`] = `"lorem-ipsum-dolor-sit-amet-f38c58113ff7eest"`;

exports[`createReplyPermlink (1)  1`] = `"re-good-karma-2018921t150500z"`;

exports[`extractMetadata (1)  1`] = `
Object {
  "image": Array [
    "http://www.xx.com/a.png",
  ],
  "users": Array [
    "lorem",
    "ipsum",
  ],
}
`;

exports[`extractMetadata (2)  1`] = `
Object {
  "image": Array [
    "http://www.xx.com/a.png",
    "https://img.esteem.ws/h74zrad2fh.jpg",
  ],
  "links": Array [
    "http://www.google.com/foo/bar",
  ],
  "users": Array [
    "lorem",
    "ipsum",
  ],
}
`;

exports[`makeJsonMetadata (1)  1`] = `
Object {
  "app": "esteem/2.0.0-surfer",
  "community": "esteem.app",
  "format": "markdown+html",
  "image": Array [
    "http://www.xx.com/a.png",
    "https://img.esteem.ws/h74zrad2fh.jpg",
  ],
  "links": Array [
    "http://www.google.com/foo/bar",
  ],
  "tags": Array [
    "esteem",
    "art",
  ],
  "users": Array [
    "lorem",
    "ipsum",
  ],
}
`;

exports[`makeJsonMetadataReply (1) 1`] = `
Object {
  "app": "esteem/undefined-surfer",
  "community": "esteem.app",
  "format": "markdown+html",
  "tags": Array [
    "foo",
    "bar",
  ],
}
`;

exports[`makeOptions (1) Default 50% / 50% 1`] = `
Object {
  "allow_curation_rewards": true,
  "allow_votes": true,
  "author": "talhasch",
  "extensions": Array [
    Array [
      0,
      Object {
        "beneficiaries": Array [
          Object {
            "account": "esteemapp",
            "weight": 300,
          },
        ],
      },
    ],
  ],
  "max_accepted_payout": "1000000.000 SBD",
  "percent_steem_dollars": 10000,
  "permlink": "lorem-ipsum-1",
}
`;

exports[`makeOptions (2) Power Up 100% 1`] = `
Object {
  "allow_curation_rewards": true,
  "allow_votes": true,
  "author": "talhasch",
  "extensions": Array [
    Array [
      0,
      Object {
        "beneficiaries": Array [
          Object {
            "account": "esteemapp",
            "weight": 300,
          },
        ],
      },
    ],
  ],
  "max_accepted_payout": "1000000.000 SBD",
  "percent_steem_dollars": 0,
  "permlink": "lorem-ipsum-1",
}
`;

exports[`makeOptions (3) Decline Payout 1`] = `
Object {
  "allow_curation_rewards": true,
  "allow_votes": true,
  "author": "talhasch",
  "extensions": Array [
    Array [
      0,
      Object {
        "beneficiaries": Array [
          Object {
            "account": "esteemapp",
            "weight": 300,
          },
        ],
      },
    ],
  ],
  "max_accepted_payout": "0.000 SBD",
  "percent_steem_dollars": 10000,
  "permlink": "lorem-ipsum-1",
}
`;
